### PR TITLE
Clean up build

### PR DESCRIPTION
**Pull request by basil**

### CODE REVIEW

File: plugin-management-cli/pom.xml [hunk 1]
```diff
@@ -24,9 +24,15 @@
...
-            <version>3.0.2</version>
+            <groupId>com.github.spotbugs</groupId>
+            <artifactId>spotbugs-annotations</artifactId>
+            <optional>true</optional>
+            <exclusions>
+                <exclusion>
+                    <groupId>com.google.code.findbugs</groupId>
+                    <artifactId>jsr305</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (basil) [inline plugin-management-cli/pom.xml:35]: EOL JSR 305 as we have in other repository; switch to SpotBugs JAR.

File: plugin-management-library/pom.xml [hunk 2]
```diff
@@ -68,45 +68,37 @@
         <dependency>
             <groupId>com.github.spotbugs</groupId>
             <artifactId>spotbugs-annotations</artifactId>
+            <optional>true</optional>
+            <exclusions>
+                <exclusion>
+                    <groupId>com.google.code.findbugs</groupId>
+                    <artifactId>jsr305</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (basil) [inline plugin-management-library/pom.xml:77]: Consistent with other repositories, ban JSR 305.

File: plugin-management-library/pom.xml [hunk 3]
```diff
@@ -68,45 +68,37 @@
...
-             <groupId>com.github.stefanbirkner</groupId>
-             <artifactId>system-rules</artifactId>
-             <version>1.19.0</version>
-             <scope>test</scope>
-         </dependency>
+            <groupId>com.github.stefanbirkner</groupId>
+            <artifactId>system-rules</artifactId>
+            <version>1.19.0</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (basil) [inline plugin-management-library/pom.xml:102]: Fixing incorrect indentation.

File: plugin-management-library/pom.xml [hunk 4]
```diff
@@ -68,45 +68,37 @@
...
-                    <groupId>org.apache.maven.plugins</groupId>
-                    <artifactId>maven-javadoc-plugin</artifactId>
-                    <configuration>
-                        <source>8</source>
-                    </configuration>
-                </plugin>
-            </plugins>
-        </pluginManagement>
-    </build>
-
```
Comment (basil) [inline plugin-management-library/pom.xml:111]: Redundant; already in the grandparent POM.

File: pom.xml [hunk 5]
```diff
@@ -25,18 +27,17 @@
         <module>plugin-management-library</module>
         <module>plugin-management-cli</module>
     </modules>
-
-    <scm>
-        <connection>scm:git:ssh://git@github.com/jenkinsci/plugin-installation-manager-tool.git</connection>
-        <developerConnection>scm:git:ssh://git@github.com/jenkinsci/plugin-installation-manager-tool.git</developerConnection>
-        <url>https://github.com/jenkinsci/plugin-installation-manager-tool</url>
+    <scm child.scm.connection.inherit.append.path="false" child.scm.developerConnection.inherit.append.path="false" child.scm.url.inherit.append.path="false">
```
Comment (basil) [inline pom.xml:30]: Good for multi-module Maven projects.

File: pom.xml [hunk 6]
```diff
@@ -25,18 +27,17 @@
...
+        <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
+        <developerConnection>scm:git:git@github.com:${gitHubRepo}.git</developerConnection>
+        <url>https://github.com/${gitHubRepo}</url>
```
Comment (basil) [inline pom.xml:33]: Incrementalified.

File: pom.xml [hunk 7]
```diff
@@ -25,18 +27,17 @@
...
         <tag>${scmTag}</tag>
     </scm>
 
     <licenses>
         <license>
             <name>The MIT license</name>
-            <url>http://www.opensource.org/licenses/mit-license.php</url>
+            <url>https://opensource.org/licenses/MIT</url>
```
Comment (basil) [inline pom.xml:40]: Current URL with HTTPS.

File: pom.xml [hunk 8]
```diff
@@ -57,17 +58,6 @@
...
-                <groupId>junit</groupId>
-                <artifactId>junit</artifactId>
-                <version>4.13.2</version>
-                <exclusions>
-                    <exclusion>
-                        <groupId>org.hamcrest</groupId>
-                        <artifactId>hamcrest-core</artifactId>
-                    </exclusion>
-                </exclusions>
-            </dependency>
```
Comment (basil) [inline pom.xml:70]: Redundant; in the parent POM.

File: pom.xml [hunk 9]
```diff
@@ -78,13 +68,6 @@
                 <artifactId>hamcrest-core</artifactId>
                 <version>2.2</version>
             </dependency>
-            <dependency>
-                <groupId>org.junit</groupId>
-                <artifactId>junit-bom</artifactId>
-                <version>5.9.0</version>
-                <type>pom</type>
-                <scope>import</scope>
-            </dependency>
```
Comment (basil) [inline pom.xml:87]: Redundant; in the parent POM.

**Review by timja — APPROVED** [review]

**Pull request merged**